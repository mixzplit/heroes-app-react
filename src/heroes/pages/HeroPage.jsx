import { useMemo } from "react";
import { Navigate, useNavigate, useParams } from "react-router-dom"
import { getHeroById } from "../helpers";

export const HeroPage = () => {

  const {id } = useParams();
  const navigate = useNavigate();

  const hero = useMemo( () => getHeroById(id), [id]);


  const onNavigateBack = () => {
    navigate(-1);
  }
  /** Validamos que venga un Heroe valido */
  if( !hero ) {
    return <Navigate to="/marvel" />
  }

  return (
    <div className="row mt-4 mb-3">
      <div className="col-4">
        <img src={`/assets/heroes/${id}.jpg`} alt={hero.superhero} className="img-thumbnail animate__animated animate__fadeInLeft" />
      </div>

      <div className="col-8">
{/*         <h3>{hero.superhero}</h3>
        <ul className="list-group list-group-flush">
          <li className="list-group-item"><b>Alter Ego:</b> {hero.alter_ego}</li>
          <li className="list-group-item"><b>Publisher:</b> {hero.publisher}</li>
          <li className="list-group-item"><b>First Appearance:</b> {hero.first_appearance}</li>
        </ul>

        <h5 className="mt-3">Characters</h5>
        <p>{hero.characters}</p> */}

        <div className="card border-light mb-3" style={{maxWidth: 780 }}>
          <div className="card-header">{hero.superhero}</div>
          <div className="card-body">
            <h5 className="card-title">Dark card title</h5>
            <ul className="list-group list-group-flush">
              <li className="list-group-item"><b>Alter Ego:</b> {hero.alter_ego}</li>
              <li className="list-group-item"><b>Publisher:</b> {hero.publisher}</li>
              <li className="list-group-item"><b>First Appearance:</b> {hero.first_appearance}</li>
            </ul>
            <h5 className="mt-3">Characters</h5>
            <p>{hero.characters}</p>

            <button onClick={onNavigateBack} className="btn btn-outline btn-primary">Regresar</button>
          </div>
        </div>
      </div>

    </div>
  )
}
