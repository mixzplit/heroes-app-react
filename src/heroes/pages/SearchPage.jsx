import { useLocation, useNavigate } from "react-router-dom";
import queryString from 'query-string';

import { useForm } from "../../hooks";
import { HeroCard  } from "../components";

export const SearchPage = () => {

  const navigate = useNavigate();
  const location = useLocation();
  console.log({location});

  const { q = '' } = queryString.parse(location.search)

  const { searchText, onInputChange } = useForm({
    searchText: ''
  });

  const onSubmitSearch = (event) => {
    event.preventDefault()

    if(searchText.trim().length <= 1) return;

    navigate(`?q=${searchText.toLowerCase()}`)

    console.log({searchText});
  }

  return (
    <>
      <h1>SearchPage</h1>
      <hr />

      <div className="row">
        <div className="col-5">
          <h4>Searching</h4>
          <hr />
          <form action="" onSubmit={onSubmitSearch}>
            <input type="text" placeholder="Search a hero" 
              name="searchText" 
              id="searchText" 
              className="form-control"
              autoComplete="off"
              value={searchText}
              onChange={ onInputChange } />
            <button className="btn btn-outline-primary mt-1">Search</button>
          </form>
        </div>
        
        <div className="col-7">
          <h4>Results</h4>
          <hr />

          <div className="alert alert-primary">
            Search a hero
          </div>
          <div className="alert alert-danger">
            No hero with <b>{ q }</b>
          </div>

          {/* <HeroCard /> */}

        </div>

      </div>


    </>
  )
}
