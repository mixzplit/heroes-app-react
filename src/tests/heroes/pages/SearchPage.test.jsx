import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../heroes/pages/SearchPage";

describe('Pruebas en <SearchPage />', () => { 

  test('should show values default correctly', () => { 
      const {container} = render(
        <MemoryRouter>
          <SearchPage />
        </MemoryRouter>
      );

      expect(container).toMatchSnapshot();
   }); 

 })