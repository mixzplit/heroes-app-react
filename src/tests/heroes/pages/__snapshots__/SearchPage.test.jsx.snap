// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> should show values default correctly 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        action=""
      >
        <input
          autocomplete="off"
          class="form-control"
          id="searchText"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a hero
         
      </div>
      <div
        class="alert alert-primary"
        style="display: none;"
      >
        No hero with 
         
      </div>
    </div>
  </div>
</div>
`;
